ols/app/endpoints/ols.py

```diff
@@ -8,16 +8,39 @@
 from ols.app.models.models import LLMRequest
 from ols.app.utils import Utils
 from ols.src.docs.docs_summarizer import DocsSummarizer
-from ols.src.llms.llm_loader import LLMLoader
+from ols.src.llms.llm_loader import LLMConfigurationError, LLMLoader
 from ols.src.query_helpers.question_validator import QuestionValidator
 from ols.src.query_helpers.yaml_generator import YamlGenerator
 from ols.utils import config
 
 router = APIRouter(tags=["query"])
 
 
+# NOTE: As we are using default values for provider and model in "query helpers",
+# we need to catch the case when user provides only provider and not model
+# (and vice-versa). Eventually, we should construct proper config object from
+# user's input and use its verification capabilites.
+def verify_request_provider_and_model(llm_request: LLMRequest) -> None:
+    """Verify that the provider and model are set in the request."""
+    # exclusive or - both values needs to be set
+    if bool(llm_request.provider) != bool(llm_request.model):
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail={
+                "response": "Please provide LLM provider and model into "
+                "your request. Both values needs to be present."
+            },
+        )
+
+    if llm_request.model and llm_request.provider:
+        config.default_logger.info(
+            f"provider '{llm_request.provider}' is set in request"
+        )
+        config.default_logger.info(f"model '{llm_request.model}' is set in request")
+
+
 @router.post("/query")
-def conversation_request(llm_request: LLMRequest) -> LLMRequest:
+def conversation_request(llm_request: LLMRequest) -> LLMRequest:  # noqa: C901
```

Comment:
This `# noqa: C901` blocks the `ols/app/endpoints/ols.py:43:5: C901 conversation_request is too complex (11 > 10)` linting error.

The current code flow is not ideal. It's not very readable and the endpoint contains a lot of logic. Consider function flow like
```python
@router.post("/query")
def conversation_request(llm_request: LLMRequest) -> LLMRequest:
   conversation_id = resolve_conversation_id(llm_request)
   validation_result = validate_query(llm_request)
   if validation_result…
   …
```
or even handle just the REST API logic in the endpoint and leave the responsibilities to other relevant parts like
```python
def conversation_request(llm_request: LLMRequest) -> LLMRequest:
   # Validate the request
    if not service.validate_request(request):
        raise HTTPException(status_code=400, detail="Invalid request")

    # Process the request
    result = process_request(request)

    # Return the response
    return ResponseModel(result=result)
```
With these changes, it can be covered also by unit tests, so we don't need to test every scenario in integration tests.

But I wouldn't do that in this PR. I can either do that before this PR or after. WDYT?

Reply:
yup. Later it would be valuable to split the processing from REST API handling